pkg/extensions/unpack.go

```diff
@@ -0,0 +1,295 @@
+		return "", err
+	}
+
+	// read the install file information
+	installObj := apps.Deployment{}
+	if err := readFileIntoObject(fs, root, installFileName, true, &installObj); err != nil {
+		return "", err
+	}
+	extensionRecord.Spec.Controller.Deployment = &v1alpha1.ControllerDeployment{
+		Name: installObj.GetName(),
+		Spec: installObj.Spec,
+	}
+
+	// read the app file information
+	appObj := v1alpha1.AppMetadataSpec{}
+	if err := readFileIntoObject(fs, root, appFileName, true, &appObj); err != nil {
+		return "", err
+	}
+	extensionRecord.Spec.AppMetadataSpec = appObj
+
+	// read the icon file and encode to base64
+	icons, err := readIcons(fs, root)
+	if err != nil {
+		return "", err
+	}
+	extensionRecord.Spec.AppMetadataSpec.Icons = icons
+
+	// read the RBAC information
+	permissionsObj := v1alpha1.PermissionsSpec{}
+	if err := readFileIntoObject(fs, root, permissionsFileName, false /* not required */, &permissionsObj); err != nil {
+		return "", err
+	}
+	extensionRecord.Spec.Permissions = permissionsObj
+
+	// marshal the full extension record to yaml and write it to the output
+	extensionRecordRaw, err := yaml.Marshal(extensionRecord)
+	if err != nil {
+		return "", err
+	}
+	_, err = output.WriteString(yamlSeparator + string(extensionRecordRaw))
+	if err != nil {
+		return "", err
+	}
+
+	return output.String(), nil
+}
+
+func readFileIntoObject(fs afero.Fs, root, fileName string, required bool, obj interface{}) error {
+	file, err := fs.Open(filepath.Join(root, fileName))
+	if err != nil {
+		if os.IsNotExist(err) && !required {
+			// the given file doesn't exist, but it's also not required, this is OK
+			return nil
+		}
+		return err
+	}
+	defer func() { _ = file.Close() }()
+
+	b, err := ioutil.ReadAll(file)
+	if err != nil {
+		return err
+	}
+
+	if err := yaml.Unmarshal(b, obj); err != nil {
+		return fmt.Errorf("failed to unmarshal yaml from file %s: %+v", file.Name(), err)
+	}
+
+	return nil
+}
+
+func readResources(fs afero.Fs, root string) (crdList *v1alpha1.CRDList, crdContent string, err error) {
+	resourcesDir := filepath.Join(root, resourcesDirName)
+
+	// check for the existence of the resources directory
+	dirExists, err := afero.DirExists(fs, resourcesDir)
+	if !dirExists || err != nil {
+		// the resources dir doesn't appear to exist, this isn't an error but return nothing
+		return v1alpha1.NewCRDList(), "", nil
+	}
+
+	// walk the resources dir and find all the CRD files
+	resourcesFiles, err := findResourcesFiles(fs, resourcesDir)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// now that we have found all the resource files, process each one in the list
+	var sb strings.Builder
+	crdList = v1alpha1.NewCRDList()
+	for _, rf := range resourcesFiles {
+		if err := readResourceFile(fs, rf, crdList, &sb); err != nil {
+			return nil, "", err
+		}
+	}
+
+	return crdList, sb.String(), nil
+}
+
+func findResourcesFiles(fs afero.Fs, resourcesDir string) ([]string, error) {
+	resourcesFiles := []string{}
```

Comment:
Empty slice declaration via literal, consider

```suggestion
	var resourcesFiles []string
```